SigAverager/SigAverager/signal_averager_decorator.py

```diff
@@ -0,0 +1,89 @@
+"""Signal Averaging."""
+import numpy as np
+import SigAverager.cwg
+
+Number_of_iterations = 2048
+
+
+def n_iterations(n):
+    """Generate Carrier Wave with AWGN.
+
+    Parameters
+    ----------
+    n: int
+        n is the number of averages to perform
+    """
+
+    def inner(f):
+        def wrapper(*args, **kwargs):
+            rv = np.zeros(args[4])
```

Comment:
So... an implicit assumption behind decorators (which I thought was obvious but clearly I was mistaken!) is that they shouldn't need (or in fact have) any knowledge about the function to which they'll be applied.